File: Berger.Data/MsfaEntity/ELearning/QuestionSetCollection.cs
using Berger.Data.Common;

namespace Berger.Data.MsfaEntity.ELearning
{
    public class QuestionSetCollection : AuditableEntity<int>
    {
        public int QuestionSetId { get; set; }
        public QuestionSet QuestionSet { get; set; }
        public int QuestionId { get; set; }
        public Question Question { get; set; }
        public int Mark { get; set; }
    }
}

File: BergerMsfaApi/ClientApp/src/app/Shared/Entity/ELearning/questionSetCollection.ts
import { Question } from './question';
import { QuestionSet } from './questionSet';

export class QuestionSetCollection {
    id: number;
    questionSetId: number;
    questionSet: QuestionSet;
    questionId: number;
    question: Question;
    mark: number;
    status: number;

    statusText: string;
    isSelected: boolean;
    questionTitle: string;
    questionTypeText: string;
    editDeleteId: number;
    
    constructor(init?: Partial<QuestionSetCollection>) {
        Object.assign(this, init);
    }

    clear() {
        this.id = 0;
        this.questionSetId = 0;
        this.questionId = null;
        this.mark = null;
        this.status = null;
        this.statusText = '';
        this.isSelected = false;
        this.questionTitle = '';
        this.questionTypeText = '';
    }
}